.github/workflows/linux-check.yaml

```diff
@@ -98,7 +98,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        compiler: [{ CXX: g++-12, CC: gcc-12 }, { CXX: clang++-14, CC: clang-14 }]
+        compiler: [{ CXX: g++-13, CC: gcc-13 }, { CXX: clang++-15, CC: clang-15 }]
```

Comment:
Can compiler be updated independently from switching the code to C++20?

Reply:
I'd say there's no reason to update GCC in this PR and Clang 15 is still too old anyways.

Reply:
`gcc 14` and `clang 18` are now in `master`. @RedAuburn please consider a rebase

Reply:
@Ferenc- this is an incorrect statement ) `gcc 14 and clang 18 are enabled on Github CI` is the correct one.